cpp/benchmarks/transpose/transpose.cpp

```diff
@@ -40,16 +40,29 @@ static void BM_transpose(benchmark::State& state)
     cuda_event_timer raii(state, true);
     auto output = cudf::transpose(input);
   }
+
+  // Collect memory statistics.
+  auto const bytes_read    = input.num_columns() * input.num_rows() * (sizeof(int32_t));
```

Comment:
I would like to avoid potential future type mismatches that result in wrong bytes/s reports. So I think you should stash the `type_id` in a variable above:
```
constexpr auto column_type = cudf::type_id::INT32;
```

And then here use CUDF's `id_to_type` utility:
```suggestion
  auto const bytes_read    = input.num_columns() * input.num_rows() * (sizeof(cudf::id_to_type(column_type)));
```

See https://docs.rapids.ai/api/libcudf/stable/group__utility__dispatcher#gad7e12b8accf60e7c0e500294e1ee8536